src/m4st/translate/model.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABC, abstractmethod
+
+from transformers import T5ForConditionalGeneration, T5Tokenizer
+
+from m4st.ollama.client import generate
+
+language_iso_to_name = {
+    "eng": "English",
+    "deu": "German",
+    "fra": "French",
+    "jpn": "Japanese",
+    "spa": "Spanish",
+    "rou": "Romanian",
+    "zho": "Chinese",
+}
+
+
+class TranslationModel(ABC):
+    @abstractmethod
+    def __call__(self, text: str):
+        pass
+
+
+class T5TranslateModel(TranslationModel):
+    def __init__(
+        self,
+        source_lang_iso: str,
+        target_lang_iso: str,
+        model_tag: str = "google-t5/t5-small",
+    ):
+        self.model = T5ForConditionalGeneration.from_pretrained(model_tag)
+        self.tokenizer = T5Tokenizer.from_pretrained(model_tag)
+
+        self.supported_languages: list[str] = ["eng", "deu", "fra"]
+
+        assert target_lang_iso in self.supported_languages, f"This model \
+only supports {self.supported_languages}, but got target {target_lang_iso}."
+        assert source_lang_iso in self.supported_languages, f"This model \
+only supports {self.supported_languages}, but got source {source_lang_iso}."
+
+        self.source_lang_iso: str = source_lang_iso
+        self.target_lang_iso: str = target_lang_iso
+
+    def __call__(self, text: str):
+        source_lang_name, target_lang_name = (
+            language_iso_to_name[self.source_lang_iso],
+            language_iso_to_name[self.target_lang_iso],
+        )
+        model_input_text = f"translate {source_lang_name} to {target_lang_name}: {text}"
```

Comment:
should the prompt template also be an arg or is this the fixed standard for T5?

Reply:
I think it's fixed, that's what T5 was trained with for translation, as far as I understand.